backend/app/models/user.py

```diff
@@ -0,0 +1,22 @@
+from enum import StrEnum
+
+from sqlalchemy import func
+from sqlalchemy.orm import Mapped, mapped_column
+
+from app.models.base import Base
+
+
+class UserRole(StrEnum):
+    STUDENT = "STUDENT"
+    LECTURER = "LECTURER"
+
+
+class User(Base):
+    __tablename__ = "user"
+
+    id: Mapped[str] = mapped_column(primary_key=True, server_default=func.gen_random_uuid())
+    name: Mapped[str]
+    email: Mapped[str] = mapped_column(unique=True)
+    password: Mapped[str]
+    role: Mapped[UserRole]
```

Comment:
I guess we won't be downgrading db, so it's just a funfact 😄 . In postgresql, by default this enum creates Type on DB level, which alembic does not see (alembic downgrade migration only drops table). And in case of downgrade-upgrade flow i believe postgres might raise exception that a type already exists. Nice workaround is to add `native_enum=False` to `mapped_column` Enum definiton so Postgres doesn't create this type explicitly.